drivers/syslog/syslog_devchannel.c

```diff
@@ -169,9 +135,11 @@ int syslog_dev_channel(void)
       return -ENOMEM;
     }
 
-  /* Register the channel operations */
+  /* Register any custom channel operations */
 
-  g_syslog_dev_channel->sc_ops = &g_syslog_ops;
+#ifdef CONFIG_SYSLOG_CHAR_CRLF
+  g_syslog_dev_channel->sc_ops->sc_putc = syslog_devchan_putc;
```

Comment:
could remove syslog_devchan_putc to since syslogstream_putc already append '\r' before '\n'